speechbrain/dataio/audio_io.py

```diff
@@ -0,0 +1,232 @@
+"""
+Lightweight soundfile-based audio I/O compatibility layer.
+
+This module provides a minimal compatibility wrapper for audio I/O operations
+using soundfile (pysoundfile) library, replacing torchaudio's load, save, and
+info functions.
+
+Authors
+ * SpeechBrain Contributors 2025
+"""
+
+import numpy as np
+import soundfile as sf
+import torch
+
+
+class AudioInfo:
+    """Container for audio file metadata, compatible with torchaudio.info output.
+    
+    Attributes
+    ----------
+    sample_rate : int
+        Sample rate of the audio file.
+    frames : int
+        Total number of frames in the audio file.
+    channels : int
+        Number of audio channels.
+    subtype : str
+        Audio subtype/encoding (e.g., 'PCM_16', 'PCM_24').
+    format : str
+        Container format (e.g., 'WAV', 'FLAC').
+    duration : float
+        Duration in seconds.
+    """
+    
+    def __init__(self, sample_rate, frames, channels, subtype, format):
+        self.sample_rate = sample_rate
+        self.frames = frames
+        self.num_frames = frames  # Alias for compatibility
+        self.channels = channels
+        self.num_channels = channels  # Alias for compatibility
+        self.subtype = subtype
+        self.format = format
+    
+    @property
+    def duration(self):
+        """Calculate duration in seconds."""
+        return self.frames / self.sample_rate if self.sample_rate > 0 else 0.0
+    
+    def __repr__(self):
+        return (
+            f"AudioInfo(sample_rate={self.sample_rate}, frames={self.frames}, "
+            f"channels={self.channels}, subtype='{self.subtype}', "
+            f"format='{self.format}', duration={self.duration:.2f}s)"
+        )
+
+
+def load(path, *, channels_first=True, dtype=torch.float32, always_2d=False, 
```

Comment:
The `always_2d` argument should be True by default to match the `torchaudio.load` behavior.